### ISSUE TITLE
Request: order of TO&E

### ISSUE THREAD

**Issue by binaryspica** [MegaMek/mekhq#1535]
### Environment
Ver 0.47.4

### Description
For now TO&E looks like...

Company
├[Lance 01]
│├Unit 01-01
│├Unit 01-02
│├Unit 01-03
│└Unit 01-04
├[Lance 02]
│├[Team 02-03]
││├OmniMech 02-03
││└Battle Armor 02-03-01
│├Unit 02-01
│├Unit 02-02
│└Unit 02-04
└[Lance 03]
□├[Infantry Platoon 03-04]
□│├Squad 01
□│├Squad 02
□│├Squad 03
□│└Platoon HQ
□├Unit 03-01
□├Unit 03-02
□├Unit 03-03
□└Infantry Transport 03-04

Although it is good already, but I have some problems.

First, what about to change the order freely, rather than always put the sub-force top of the list(and order by rank)? On this example company, Team 02-03 on Lance 02 and Infantry Platoon 03-04 are must to be on the top of the list of Lance, despite their number suggests that they are must be below of 02-02/03-03.

Second, what about to put the list BELOW of the element, not just the force? In this case, put the Infantry Platoon below of Infantry Transport 03-04, and move OmniMech 02-03 on Lance 02 then place Battle Armor 02-03-01 to below of OmniMech 02-03 would be much easy to see.

Company
├[Lance 01]
│├Unit 01-01
│├Unit 01-02
│├Unit 01-03
│└Unit 01-04
├[Lance 02]
│├Unit 02-01
│├Unit 02-02
│├OmniMech 02-03
││└Battle Armor 02-03-01
│└Unit 02-04
└[Lance 03]
□├Unit 03-01
□├Unit 03-02
□├Unit 03-03
□└Infantry Transport 03-04
□□└[Infantry Platoon 03-04]
□□□├Squad 01
□□□├Squad 02
□□□├Squad 03
□□□└Platoon HQ

So what I mean is make it to look like this.

Finally, when using these forces on MegaMek, what about to define the order of the unit same as the order of the unit on TO&E?

I don't think that it would be implemented near future, and I don't expect that it will have the priority over the other urgent matters. Please, just think about that once when you have some time to touch TO&E. 

An alternative idea to implement this is, make an option to put the portrait of one of the element rather than only put the icon on the force would makes the similar result, for you can spamming sub-forces to make the order as you want.

Edit: also... it would be better if the order also affects sort by Force on Personnel/Hanger tab.

**Comment by Windchild292** [comment]
This can be done by drag and drop already, and was likely blocked by either a bug or Java issues. Closing until proven elsewise.

**Comment by binaryspica** [comment]
> This can be done by drag and drop already, and was likely blocked by either a bug or Java issues. Closing until proven elsewise.

No, it is not possible at all. You cannot put an individual unit above of the force, neither you cannot assign a unit below of an another unit, which are the core of the request.

**Comment by sixlettervariables** [comment]
I still cannot reorder the TOE in any way on Java 11.

But, you will not be able to order units above a Force due to a Java limitation.

Recording units is a possibility.

**Comment by IllianiCBT** [comment]
This RFE is blocked by technical issues and functions added in 2024 rely on static TO&E ordering. Closing.
